it/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesControllerIntegrationSpec.scala

```diff
@@ -130,144 +161,251 @@ class MessagesControllerIntegrationSpec
-        s"/traders/GB0123456789/movements/departures/${movementId.value}/messages/${messageId.value}",
-        FakeHeaders(
-          Seq(
-            "x-message-type"         -> "IE015",
-            "x-request-id"           -> UUID.randomUUID().toString,
-            HeaderNames.CONTENT_TYPE -> MimeTypes.XML
-          )
-        ),
-        Source.single(ByteString(sampleOutgoingXml))
-      )
+        val apiRequest = FakeRequest(
+          "POST",
+          s"/traders/GB0123456789/movements/departures/${movementId.value}/messages/${messageId.value}",
+          FakeHeaders(
+            Seq(
+              "x-message-type"         -> "IE015",
+              "x-request-id"           -> UUID.randomUUID().toString,
+              HeaderNames.CONTENT_TYPE -> MimeTypes.XML
+            )
+          ),
+          Source.single(ByteString(sampleOutgoingXml))
+        )
 
-      running(newApp) {
-        val sut    = newApp.injector.instanceOf[MessagesController]
-        val result = sut.outgoing(EoriNumber("GB0123456789"), MovementType("departures"), movementId, messageId)(apiRequest)
+        running(newApp) {
+          val sut    = newApp.injector.instanceOf[MessagesController]
+          val result = sut.outgoing(EoriNumber("GB0123456789"), MovementType("departures"), movementId, messageId)(apiRequest)
 
-        Helpers.status(result) mustBe ACCEPTED
+          Helpers.status(result) mustBe CREATED
+        }
       }
-    }
-
-    "with a valid body routing to XI, return 202" in {
-      // We do this instead of using the standard "app" because we otherwise get the error
-      // "Trying to materialize stream after materializer has been shutdown".
-      // We suspect it's due to nested tests.
-      val newApp                  = appBuilder.build()
-      val conversationId          = ConversationId(UUID.randomUUID())
-      val (movementId, messageId) = conversationId.toMovementAndMessageId
 
-      val time      = OffsetDateTime.of(2023, 2, 14, 15, 55, 28, 0, ZoneOffset.UTC)
-      val formatted = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH).withZone(ZoneOffset.UTC).format(time)
+      "with a small valid body routing to XI, return 202" in {
+        // We do this instead of using the standard "app" because we otherwise get the error
+        // "Trying to materialize stream after materializer has been shutdown".
+        // We suspect it's due to nested tests.
+        val newApp                  = appBuilder.build()
+        val conversationId          = ConversationId(UUID.randomUUID())
+        val (movementId, messageId) = conversationId.toMovementAndMessageId
+
+        val time      = OffsetDateTime.of(2023, 2, 14, 15, 55, 28, 0, ZoneOffset.UTC)
+        val formatted = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH).withZone(ZoneOffset.UTC).format(time)
+
+        // should not happen
+        server.stubFor(
+          post(
+            urlEqualTo("/xi")
```

Comment:
same here please add withRequestBody

Reply:
That said, the comment is wrong, so that needs fixing